pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -115,144 +257,249 @@ void applyHadamard(
     const std::size_t num_qubits, const std::vector<std::size_t> &wires,
     [[maybe_unused]] const bool inverse = false,
     [[maybe_unused]] const std::vector<PrecisionT> &params = {}) {
-    applyNC1Functor(
-        ExecutionSpace{}, arr_, num_qubits, wires,
-        KOKKOS_LAMBDA(Kokkos::View<Kokkos::complex<PrecisionT> *> arr,
-                      const std::size_t i0, const std::size_t i1) {
-            [[maybe_unused]] const auto i0_ = i0;
-            const Kokkos::complex<PrecisionT> v0 = arr(i0);
-            const Kokkos::complex<PrecisionT> v1 = arr(i1);
-            arr(i0) = M_SQRT1_2 * v0 +
-                      M_SQRT1_2 * v1; // NOLINT(readability-magic-numbers)
-            arr(i1) = M_SQRT1_2 * v0 +
-                      -M_SQRT1_2 * v1; // NOLINT(readability-magic-numbers)
-        });
+    applyNCHadamard<ExecutionSpace, PrecisionT>(arr_, num_qubits, {}, {}, wires,
+                                                inverse);
 }
 
 template <class ExecutionSpace, class PrecisionT>
-void applyS(Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
-            const std::size_t num_qubits, const std::vector<std::size_t> &wires,
-            const bool inverse = false,
-            [[maybe_unused]] const std::vector<PrecisionT> &params = {}) {
+void applyNCS(Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
+              const std::size_t num_qubits,
+              const std::vector<std::size_t> &controlled_wires,
+              const std::vector<bool> &controlled_values,
+              const std::vector<std::size_t> &wires, const bool inverse = false,
+              [[maybe_unused]] const std::vector<PrecisionT> &params = {}) {
     const Kokkos::complex<PrecisionT> shift =
         (inverse) ? Kokkos::complex<PrecisionT>{0.0, -1.0}
                   : Kokkos::complex<PrecisionT>{0.0, 1.0};
-    applyNC1Functor(
-        ExecutionSpace{}, arr_, num_qubits, wires,
+    auto core_function =
         KOKKOS_LAMBDA(Kokkos::View<Kokkos::complex<PrecisionT> *> arr,
                       const std::size_t i0, const std::size_t i1) {
-            [[maybe_unused]] const auto i0_ = i0;
-            arr(i1) *= shift;
-        });
+        [[maybe_unused]] const auto i0_ = i0;
```

Comment:
Idem.

Reply:
will fix in later commit (issue with GCC/Clang)